app/services/liquid_assigns_service.rb

```diff
@@ -15,4 +18,45 @@ def assigns
 
   def assigns_pro
   end
+
+  # This method uses Liquid::VariableLookup to find all liquid variables from
+  # a given text. We use the list to know which project assign we need.
+  def assigns_from_content
+    return AVAILABLE_PROJECT_ASSIGNS if text.nil?
+
+    variable_lookup = Liquid::VariableLookup.parse(text)
+    return (variable_lookup.lookups & AVAILABLE_PROJECT_ASSIGNS)
+  end
+
+  def cached_drops(records)
+    return [] if records.empty?
+
+    type = records.first.class.to_s.underscore
```

Comment:
What about having `type` as a parameter to save us this line since we already have the value we need in `project_assigns` and it would simplify this method a little